File: src/main/java/practice/common/dto/CommonResponse.java
package practice.common.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

@Getter
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CommonResponse<T> {

  @Builder.Default
  String code = ResponseCode.Success.name();

  @Builder.Default
  String message = ResponseCode.Success.getMessage();

  T detail;

}

File: src/main/kotlin/practice/common/dto/CommonResponse.kt
package practice.common.dto

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class CommonResponse<T>(
    val code: String = ResponseCode.Success.name,
    val message: String = ResponseCode.Success.message,
    val detail: T? = null
)
